app/assets/javascripts/discourse/components/d-editor.js.es6

```diff
@@ -636,23 +638,44 @@ export default Ember.Component.extend({
     return null;
   },
 
+  @computed
+  pastePlaceholder() {
+    return I18n.t('pasting');
+  },
+
   paste(e) {
     const clipboard = clipboardData(e);
     const types = clipboard.types;
-    let preventDefault = false;
+    const plainText = clipboard.getData("text/plain");
+    const placeholder = plainText || this.get('pastePlaceholder');
+    let handled = false;
 
-    if (types.some(t => t === "text/plain")) {
-      const text = clipboard.getData("text/plain");
-      const table = this._detectTable(text);
+    if (plainText) {
+      const table = this._detectTable(plainText);
       if (table) {
         this._addText(this._getSelected(), table);
-        preventDefault = true;
+        handled = true;
       }
-    } else if (types.some(t => t === "Files")) {
-      preventDefault = true;
     }
 
-    if (preventDefault) {
+    if (!handled && types.some(t => t === "text/html")) {
```

Comment:
Wouldn’t it be better to use includes here ?

```
types.includes("text/html")
```

It looks to me that `some` allows you to provide a custom function if you have a more complicated logic than a `===`

note there's at least another occurrence of some later